docs/governance/contribute/git-guideline.md

```diff
@@ -0,0 +1,109 @@
+
+在选择合适的模版后，只需要填写模版上的要求填写的内容即可。如果在创建issue的时候发现没有模版，或者模版内容为空，可以通过微信群，钉钉群或者邮件向我们反馈这个问题。
+
+## 2. 关于Git分支
+我们参考[Angular规范](https://github.com/angular/angular/blob/main/CONTRIBUTING.md)为分支命名。
+推荐的格式如下：
+```
+{type}/{your_account_ID}/{a_short_description}
+```
+分支名称主要包括三个字段，并通过 “/” 分割。其中：
+ - {type} : 当前分支内容的类型。
+ - {your_account_ID}: 进行这次修改的账户ID。
+ - {a_short_description}: 一个简短的描述，介绍这个分支的主要内容。
+
+ e.g. 张三创建了一个修复了输出格式化bug的分支，并打算使用账户 zhangS 提交 mr，对应的分支名称`fix/zhangS/fix_output_fmt_bug`。
+
+## 3. 关于Git Commit
+我们参考[Commitizen](https://github.com/commitizen/cz-cli)书写Commit Message。
+```
+注: 如果直接使用Commitizen生成Commit Message，需要注意因为Commitizen是开发人员
+管理commit的工具，与项目本身无关联，因此由Commitizen生成的中间产物(如: node_modules文件
+目录)可能没有在项目.gitignore文件中。
+```
+如果手动编写Commit Message，我们也建议采用[Commitizen](https://github.com/commitizen/cz-cli)的commit message格式。
+
+```
+{type} ( {component_or_file} ) {a_short_description}
+    {a_longer_description}
+    BREAKING CHANGE: {breaking_change_description}.
+    {linked issue}
+```
+
+其中主要包括6个字段：
+ - {type} : 当前commit对应的分支的类型。
+ - {component_or_file}: 当前commit改动的模块或者文件的名称。
+ - {a_short_description}: 简短的描述介绍commit中的内容。
+ - {a_longer_description}: 详细的描述用来介绍commit中的内容。
+ - {breaking_change_description}: 如果commit中包含破环兼容性的改动，需要对兼容性改动产生的影响进行介绍。
+ - {linked issue}: 与当前这个commit关联的issue。 
+
+  其中{breaking_change_description}和{linked issue}如果commit中不包含破坏兼容性的改动和关联的issue，可以省略。
+
+  e.g. 张三在分支`fix/zhangsan/fix_output_fmt_bug`中创建的commit。
+  ```
+
+    fix(kclvm-printer): fix an output format bug in kclvm-printer
+    
+    There is an output format bug in kclvm-printer because ...,
+    So, The calling of method "XXX" is replaced by "HHHH"...,
+    ...
+    
+    -- 如果没有破坏兼容性的改动和关联的issue可以省略下面内容。
+
+    BREAKING CHANGE: This change maybe cause .......
+    
+    fix #123
+
+  ```
+
+## 4. 关于 pull request
+
+在提交一个PR之前，可能需要优先考虑以下几个问题:
+- 请先查阅已经关闭的PR，也许在已经关闭的PR中，可能存在已经完成的解决方案。
+- 我们建议在提交变更之前，提交一个对应的issue描述变更中将要解决的问题，并将变更对应的PR与issue关联。
+- 在向我们提交PR之后，请签署[Contributor License Agreement(CLA)](#cla)，如果拒绝签署，我们将无法接受PR。
+- 请确保每次改动都创建了一个新的分支，并根据上文中提到的规范为分支命名。
+- 一次PR请不要超过两个commit，请将多余的commit通过squash压缩，并根据上文中提到的规范，编写commit message。
+- 我们提供了 [PR 模版](https://github.com/KusionStack/.github/blob/main/.github/PULL_REQUEST_TEMPLATE.md)，只需要添加模版中要求的内容即可，如果在创建PR时发现没有模版或者模版内容为空，可以通过微信群，钉钉群或者邮件向我们反馈这个问题。
+
+我们建议PR的标题与分支名、commit message风格保持一致：
+```
+{type} ( {component_name_or_file_name} ) :{a_short_description}
+```
+
+e.g. 张三为分支`fix/zhangsan/fix_output_fmt_bug`创建的PR名称。
+```
+fix(kclvm-printer): fix an output format bug in kclvm-printer.
+```
+
+## 5. 目前type支持的类型
+```
+- feat:     -- 添加了新的功能特性。
+- fix:      -- 进行了Bug的修复。
+- docs:     -- 进行了文档部分的修改。
+- style:    -- 对代码格式的修改，并不影响代码的功能，如：删除多余空格，代码缩进等。
+- refactor: -- 在不改变代码功能的基础上对代码进行了的重构。
+- perf:     -- 对代码进行了性能优化。
+- test:     -- 添加或者调整已有的测试用例。
+- build:    -- 对构建系统或者外部依赖库进行了调整。
+- ci:       -- 调整了CI的配置文件或者脚本。
+- chore:    -- 对源代码和测试文件之外其他部分的调整。
+- revert:   -- 对commit进行回滚。
+```
```

Comment:
Small tip: Suggestion is attached to the reference link:
[Git Commit Guidelines](https://github.com/angular/angular.js/blob/master/DEVELOPERS.md#-git-commit-guidelines)